app/controllers/AccountManagement.scala

```diff
@@ -132,6 +133,28 @@ object ManageDelivery extends ContextLogging {
     }
   }
 
+  def raiseNonDeliveryCase(deliveryIssue: DeliveryIssue)(implicit request: Request[DeliveryIssue], touchpoint: TouchpointBackend.Resolution): Future[Result] = {
+    implicit val tpBackend = touchpoint.backend
+    logger.info(s"Delivery issue is: $deliveryIssue")
+    val futureResponse = FulfilmentLookupService.lookupSubscription(tpBackend.environmentName, deliveryIssue)
+    futureResponse.map { response =>
+      val responseBody = response.body().string()
+      response.body.close()
```

Comment:
I'd do all the nuts and bolts returning something nice in the lookupSubscription call, and just return a Future[boolean] or something to say was it successful or not.
Otherwise the fact that it's an http call leaks into the calling code.